File: solutions/88-musfik/AgeOfVillagers/Nation/EgyptianKings.cs
using System;
using System.Drawing;

namespace AgeOfVillagers
{
    public class EgyptianKings : INation
    {
        private readonly Color color = Color.Yellow;
        private String nation_name = "Egyptian Kings";

        public EgyptianKings()
        {
            
        }

        public string nationName { get => nation_name; set => nation_name = value; }

        public House getHouse(Point point)
        {
            return new EgyptianHouse(point);
        }
        
        public string getNationName()
        {
            return nation_name;
        }
        

        public Color getTerrainColor()
        {
            return color;
        }

        public Tree getTree(Point point)
        {
            return new EgyptianTree(point);
        }

        public WaterResource getWaterResource(Point point)
        {
            return new EgyptianWaterResource(point);
        }
    }
}

File: solutions/72-rafidhaque/src/EgyptianKings.java
import javafx.scene.Group;
import javafx.scene.paint.Color;
import javafx.scene.shape.Rectangle;

public class EgyptianKings implements INation {
    String nationName = "Egyptian Kings";
    Group drawSpace;
    DrawHouse house;
    DrawTree tree;
    DrawWater water;

    public EgyptianKings(Group drawSpace) {
        this.drawSpace = drawSpace;
        Rectangle fillColor = new Rectangle(0,0, 600, 400);
        fillColor.setFill(Color.LIGHTYELLOW);
        drawSpace.getChildren().add(fillColor);
        house = new EgyptianKingsHouse();
        tree = new EgyptianKingsTree();
        water = new EgyptianKingsWater(12, 12);
        drawSpace.getChildren().addAll(house.draw());
        drawSpace.getChildren().addAll(tree.draw());
        drawSpace.getChildren().addAll(water.draw());
    }

    public String getNationName() {
        return nationName;
    }

    public DrawHouse getHouse() {
        return house;
    }

    public DrawTree getTree() {
        return tree;
    }

    public DrawWater getWater() {
        return water;
    }
}
